fili-core/src/main/java/com/yahoo/bard/webservice/web/endpoints/JobsServlet.java

```diff
@@ -359,20 +355,50 @@ protected void handlePreResponse(
      * notification before timeout, we retrieve the PreResponse from the PreResponseStore else we return an empty
      * Observable.
      *
-     * @param ticket  The ticket whose associated PreResponse needs to be retrieved.
-     * @param apiRequest  JobsApiRequest object with all the associated info in it
+     * @param ticket  The ticket for which the PreResponse needs to be retrieved.
+     * @param asyncAfter  The minimum duration the request is allowed to last before becoming asynchronous
+     *
+     * @return An Observable wrapping a PreResponse or an empty Observable in case a timeout occurs.
+     */
+    protected Observable<PreResponse> getResults(@NotNull String ticket, long asyncAfter) {
+        /*
+         * BroadCastChannel is a hot observable i.e. it emits notification irrespective of whether it has any
+         * subscribers. We use the replay operator so that the preResponseObservable upon connection, will begin
+         * collecting values.
+         * Once a new observer subscribes to the observable, it will have all the collected values replayed to it.
+         */
+        if (asyncAfter == JobsApiRequest.ASYNCHRONOUS_ASYNC_AFTER_VALUE) {
+            return Observable.empty();
```

Comment:
Comment on why this is good to do would be nice. (It's not immediately clear _why_ we're doing this)